static/js/portico/signup.js

```diff
@@ -133,14 +136,26 @@ $(() => {
         }
     });
 
-    const show_subdomain_section = function (bool) {
-        const action = bool ? "hide" : "show";
-        $("#subdomain_section")[action]();
-    };
+    function update_subdomain_input_requirement() {
+        // Required attribute of subdomain field should be updated only when
+        // radio button choices are present, i.e., root domain is available.
+        if ($("#realm_with_subdomain").length > 0) {
+            if ($("#realm_with_subdomain").is(":checked")) {
+                $("#id_team_subdomain").prop("required", true);
+                $("#id_team_subdomain").addClass("required");
+                $("#id_team_subdomain-error").show();
+            } else {
+                $("#id_team_subdomain").prop("required", false);
+                $("#id_team_subdomain").removeClass("required");
+                $("#id_team_subdomain-error").hide();
+            }
+        }
+    }
 
-    $("#realm_in_root_domain").on("change", function () {
-        show_subdomain_section($(this).is(":checked"));
+    $("input[name='realm_url']:radio").on("change", () => {
+        update_subdomain_input_requirement();
```

Comment:
This lambda function is unnecessary, we can directly pass `update_subdomain_input_requirement`.

Reply:
Oh, I missed that. Thanks!

Reply:
Updated.